### PR TITLE
chore: remove table properties in proto

### PR DESCRIPTION
**Pull request by st1page**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

breaking changes for
- session varible `CREATE_COMPACTION_GROUP_FOR_MV` is deprecated in this PR, but we can add it back as a meta DDL's request parameter  if we need it https://github.com/risingwavelabs/risingwave/issues/14793 
- with option `retention_seconds` I want to deprecate it because the current behavior can not maintain the stream's consistency and can bring panic... And as an undocumented feature, not many users use it.  https://github.com/risingwavelabs/risingwave/issues/14791

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: src/meta/model_v2/src/table.rs [hunk 1]
```diff
@@ -108,7 +108,6 @@ pub struct Model {
     pub distribution_key: I32Array,
     pub stream_key: I32Array,
     pub append_only: bool,
-    pub properties: Property,
```
Comment (yezizp2012) [inline src/meta/model_v2/src/table.rs:111]: Need to change definition here: https://github.com/risingwavelabs/risingwave/blob/c8030fc0f2179c563452afb1983411a70267c455/src/meta/model_v2/migration/src/m20230908_072257_init.rs#L1013

File: src/meta/model_v2/src/table.rs [hunk 2]
```diff
@@ -123,6 +122,7 @@ pub struct Model {
     pub cleaned_by_watermark: bool,
     pub description: Option<String>,
     pub version: Option<TableVersion>,
+    pub retention_seconds: u32,
```
Comment (yezizp2012) [inline src/meta/model_v2/src/table.rs:125]: Can only be defined as `i32`, and since it's not `Option` you need to define `retention_seconds` as not null in migration.

Comment (BugenZhao) [reply src/meta/model_v2/src/table.rs:125]: This reminds me of whether it's possible to integrate the test for meta store migration in CI, or if we rely on compatibility testing for such changes. cc @kwannoel

File: proto/catalog.proto [hunk 3]
```diff
@@ -314,6 +315,8 @@ message Table {
   optional string initialized_at_cluster_version = 35;
   optional string created_at_cluster_version = 36;
 
+  uint32 retention_seconds = 37;
```
Comment (BugenZhao) [inline proto/catalog.proto:318]: Better to note down here that it only applies to append-only tables.

File: src/common/src/catalog/mod.rs [hunk 4]
```diff
@@ -295,24 +294,17 @@ impl From<&TableOption> for risingwave_pb::hummock::TableOption {
...
-                Ok(retention_seconds_u32) => result.retention_seconds = Some(retention_seconds_u32),
-                Err(e) => {
-                    tracing::info!(
-                        error = %e.as_report(),
-                        "build_table_option parse option ttl_string {}",
-                        ttl_string,
-                    );
-                    result.retention_seconds = None;
+        TableOption {
+            retention_seconds: retention_seconds.and_then(|ttl| {
```
Comment (BugenZhao) [inline src/common/src/catalog/mod.rs:300]: Is it essentially `Option<NonZeroU32>`?

File: src/meta/src/stream/stream_manager.rs [hunk 5]
```diff
@@ -462,6 +455,7 @@ impl GlobalStreamManager {
             create_type,
             ddl_type,
             replace_table_job_info,
+            new_independent_compaction_group,
```
Comment (BugenZhao) [inline src/meta/src/stream/stream_manager.rs:458]: Passing a boolean flag everywhere actually looks worse to me. 😕 I guess we can at least still keep it an `Options` struct, which can be made type-safer than `HashMap<String, String>` and allow us to extend it easily in the future.

File: src/meta/model_v2/src/table.rs (hunk 2 above)
Comment (st1page) [reply src/meta/model_v2/src/table.rs:125]: I think I can change it to `Option<u32> `

File: src/meta/src/stream/stream_manager.rs (hunk 5 above)
Comment (st1page) [reply src/meta/src/stream/stream_manager.rs:458]: Yes, I want to add it later because the PR has been too large. I guess the `Option` will be `CreateStreamJobOption`?

File: src/common/src/catalog/mod.rs (hunk 4 above)
Comment (st1page) [reply src/common/src/catalog/mod.rs]: change it to `Option<u32>` now, because most of the places use the prost's struct directly.

File: src/meta/model_v2/src/table.rs (hunk 2 above)
Comment (st1page) [reply src/meta/model_v2/src/table.rs]: fixed

Comment (kwannoel) [reply src/meta/model_v2/src/table.rs]: >  or if we rely on compatibility testing for such changes.

For backwards compat testing, we run some e2e tests, re-create the cluster, and check everything works as expected. I suppose it can catch most cases.

I'm not sure if there's some corner cases that would be missed.

**Review by hzxa21 — APPROVED** [review]
LGTM for the storage part. @Li0k PTAL

File: src/meta/src/stream/stream_manager.rs (hunk 5 above)
Comment (BugenZhao) [reply src/meta/src/stream/stream_manager.rs:458]: IMO the PR can not be any larger. 🤣 Just simply replace all changes of `pub new_independent_compaction_group: bool` to a struct. To me it seems strange to have this storage-specific option aside of other general options for a streaming job.

**Review by Li0k — APPROVED** [review]
LGTM for the storage part.
